public/app/features/datasources/state/actions.test.ts

```diff
@@ -42,3 +47,40 @@ describe('Find new name', () => {
     expect(findNewName(plugins, name)).toEqual('pretty cool plugin-');
   });
 });
+
+describe('initDataSourceSettings', () => {
+  describe('when pageId is not a number', () => {
+    it('then initDataSourceSettingsFailed should be dispatched', async () => {
+      const dispatchedActions = await thunkTester({})
+        .givenThunk(initDataSourceSettings)
+        .whenThunkIsDispatched('some page');
+
+      expect(dispatchedActions).toEqual([initDataSourceSettingsFailed(new Error('Invalid ID'))]);
+    });
+  });
+
+  describe('when pageId is a number', () => {
+    it('then initDataSourceSettingsSucceeded should be dispatched', async () => {
+      const thunkMock = (): ThunkResult<void> => (dispatch: ThunkDispatch, getState) => {};
+      const dataSource = { type: 'app' };
+      const dataSourceMeta = { id: 'some id' };
+      const dependencies: InitDataSourceSettingDependencies = {
+        loadDataSource: jest.fn(thunkMock),
+        getDataSource: jest.fn().mockReturnValue(dataSource),
+        getDataSourceMeta: jest.fn().mockReturnValue(dataSourceMeta),
+        importDataSourcePlugin: jest.fn().mockReturnValue({}),
+      };
+      const state = {
+        dataSourceSettings: {},
+        datasources: {},
+      };
+      const dispatchedActions = await thunkTester(state)
+        .givenThunk(initDataSourceSettings)
+        .whenThunkIsDispatched(256, dependencies);
+
+      expect(dispatchedActions).toEqual([initDataSourceSettingsSucceeded({} as GenericDataSourcePlugin)]);
+      expect(dependencies.loadDataSource).toHaveBeenCalledTimes(1);
```

Comment:
Nit: you could make sure the rest of the dependencies are called/not called and check called with params as well.